src/layers/learning/fully_connected.cpp

```diff
@@ -240,19 +234,19 @@ void fully_connected_layer<data_layout::DATA_PARALLEL, El::Device::CPU>::bp_comp
   // Compute gradient w.r.t. linearity if needed
   optimizer* linearity_optimizer = this->m_weights[0]->get_optimizer();
   if (linearity_optimizer != nullptr) {
+    DataType dst_scale = DataType(0), gradient_scale = DataType(0);
+    auto& linearity_gradient = linearity_optimizer->get_gradient_buffer(
+      dst_scale, gradient_scale, true);
+    gradient_scale /= mini_batch_size;
```

Comment:
I am trying to reconcile how the gradient_scale is modified by the mini-batch size and the redundant size of the matrix.

Reply:
See above. It may be helpful to realize that, in nearly all our code right now, `gradient_scale` will be `1.0`.